pkg/kv/kvserver/raftlog/payload.go

```diff
@@ -50,10 +50,6 @@ func EncodeCommand(
 		if raftAdmissionMeta != nil {
 			entryEncoding = EntryEncodingSideloadedWithAC
 		}
-
-		if command.ReplicatedEvalResult.AddSSTable.Data == nil {
```

Comment:
nit: keep this check and add `&& command.ReplicatedEvalResult.RemoteFileLoc == ""`?